### PR TITLE
Feature/argument parsing and mode calc splitting

### PR DESCRIPTION
**Pull request by Thrashattack**
O que acha dessa proposta de organização dos modos de cálculo ? 
Para cada modo uma especificação diferente, ou não, de arquivo config.json é necessária. 
Por enquanto só implementei o que já existia dentro do módulo de project. 

Fiz a coleta dos argumentos  de linha de comando rumo à CLI. 
A selecao do modo e a aplicação do calculo é parametrizada com os argumentos.

### CODE REVIEW

**Comment by Thrashattack** [comment]
Fiz refactoring do que estava com var para const e let, seria bom colocar no package a engine >= 8

**Comment by Thrashattack** [comment]
@filipewvandrade

**Comment by Thrashattack** [comment]
Tem um mode `questionary.js` que é referente àquela opção `--init` que a gente pode vir a disponibilizar pra iniciar um fluxo de entrevista como npm init

**Comment by Thrashattack** [comment]
> Tem um mode `questionary.js` que é referente àquela opção `--init` que a gente pode vir a disponibilizar pra iniciar um fluxo de entrevista como npm init

Não tá toda implementada. Falta coletar o modo e estruturar de acordo.

File: index.js [hunk 1]
```diff
@@ -67,24 +71,32 @@ var template = HandleBars.compile(`
...
+try {
+	fs.writeFileSync(options.output + "/budget.html", output);
+} catch(e) {
+	fs.mkdirSync(options.output, '0777', (err) => {
+		console.error(err);
+		throw new Error('Could not create output directory : ' )
+		
+	});
+
+	fs.writeFileSync(options.output + "/budget.html", output);
```
Comment (andradefil) [inline index.js:96]: Maybe the exception catch here are not exactly the error about directory not exists. The second call to `writeFileSync` call not have any benefits being in try catch block. What I suggest here is to have together in `try` block a call to IO to check if file exists and another to create the file, and if an exception occurs you could print exception in console and raise for up.

File: package.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "argparse": "^2.0.1",
```
Comment (andradefil) [inline package.json:12]: I would like to keep this project so small and simple as possible, I think that the cli for it is not too complex for necessary add a external library to handle that. Could you do the parse of arguments no using libraries?

File: README.md [hunk 3]
```diff
@@ -35,8 +38,23 @@ A project is configured using a simple json like that
...
 After to configure, the budget is ready to build
 
+# Args
+
+The possible arguments for allspark are: 
+  - `-v` or `--version`: Prints the project version.
+  - `-c` or `--config`: Specifies the config file path. It's required.
+  - `-o` or `--output`: Specifies the output directory to write the budget.html file. Defaults to `./`.
+  - `-m` or `--mode`: Specify wich mode of budget you want. The config file should be specified in this mode. Defaults to `project`.
+  - `-s` or `--shine`: Tells allspark to enligth you with all the information of whats is being done behind the scenes (verbose). Defaults to `false`.
```
Comment (andradefil) [inline README.md:56]: Good idea but I think that is more semantic to use the word `-v --verbose` is the standard in a lot of cli tools

**Review by andradefil — COMMENTED** [review]
Good work I not like the modes because introduce a domain abstraction that not exists yet, I explained that I not agree with separate modes in the project because the each project I see each to project as separate domain from each other, I think that we can have default configuration but I'm not seeing the `modes` domain, please can go ahead to clarify me about your modeling view.

as suggestion to improve organization I think that we can put all "logic" files in a folder called `src` and index.js keep on root to initialize the app

**Comment by andradefil** [comment]
> Tem um mode `questionary.js` que é referente àquela opção `--init` que a gente pode vir a disponibilizar pra iniciar um fluxo de entrevista como npm init



> Fiz refactoring do que estava com var para const e let, seria bom colocar no package a engine >= 8

Ok go ahead with that, It's be good

**Comment by andradefil** [comment]
The feature of the questionary maybe can be renamed to `requirements`? questionary is not grammatically correctly in english. Another think, is an Issue very good and you did some awesome code adjustments well, maybe is good to separate in two different PR's because I would like to use the code refactors in the next features, and apparently is more simple to merge besides the requirements feature.

**Pull request closed without merging**